lib/stages/VisSharedMemWriter.cpp

```diff
@@ -51,8 +53,9 @@ void check_remove(std::string fname) {
 
 VisSharedMemWriter::VisSharedMemWriter(Config& config, const std::string& unique_name,
                                        bufferContainer& buffer_container) :
-    Stage(config, unique_name, buffer_container,
-          std::bind(&VisSharedMemWriter::main_thread, this)) {
+    Stage(config, unique_name, buffer_container, std::bind(&VisSharedMemWriter::main_thread, this)),
+    dropped_frame_counter(Metrics::instance().add_counter(
+        "kotekan_vissharedmemwriter_dropped_frame_total", unique_name, {"fpga_count"})) {
```

Comment:
`fpga_count` is probably not a great label to use here. Essentially each label creates an entirely new timestream  and so each event dropping frames will be in its own series. Additionally prometheus is basically giving us the time axis anyway, so we don't need to generate our own version of one (although we will lose some time resolution as it pulls even minute or so).

I would recommend dropping the FPGA count label, but adding a freq_id one similar to what visWriter does.

Reply:
You should additionally add a `reason` label to differentiate the two different reasons data can get dropped.